core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1193,8 +1193,10 @@ class DAGScheduler(
         // TODO: This will be really slow if we keep accumulating shuffle map stages
         for ((shuffleId, stage) <- shuffleToMapStage) {
           stage.removeOutputsOnExecutor(execId)
-          val locs = stage.outputLocs.map(list => if (list.isEmpty) null else list.head).toArray
-          mapOutputTracker.registerMapOutputs(shuffleId, locs, changeEpoch = true)
+          if (!runningStages.contains(stage)) {
```

Comment:
@squito for this one, dagScheduler handler ExecutorLost, it will register all `shuffleToMapStage`. Including the running MapStage, which that is not complete partial, it will cause we register partial mapStatus into MapOutputTracker.